### PR TITLE
Use FFI in rnp CLI

### PR DESCRIPTION
**Pull request by ni4**
This is a draft PR and will contain progress on moving rnp to FFI completely.
Right now it compiles but still includes some old code and doesn't pass number of tests.
Also most likely it would be split to a bunch of PRs since it also updates FFI with new functions.

### CODE REVIEW

**Comment by ni4** [comment]
Let's see whether it passes test (it does locally). There is still a lot of tests which should be checked whether/how they should be moved to the FFI API so I will be able to remove deprecated code from rnpcli.cpp.

**Comment by ronaldtse** [comment]
@ni4 looks like we're nearly good to go here :wink:

**Comment by ni4** [comment]
@ronaldtse Yeah, just few more things... :)

**Comment by ni4** [comment]
@ronaldtse @dewyatt Okay, I think this one is finally ready for review.
So after this PR all CLI will use only FFI interface for all operations.
`fficli.cpp` still borrows some defines and service functions via `utils.h` but I plan to address this via separate PRs.

**Comment by ni4** [comment]
@ronaldtse @dewyatt Could you please review this?

File: src/tests/key-unlock.cpp [hunk 1]
```diff
@@ -32,21 +32,20 @@
...
 #include "crypto/hash.h"
+#include <fstream>
+
+/* we'll need direct access to secret key's pgp_key_t, should be ok for tests */
+struct rnp_key_handle_st {
+    rnp_ffi_t        ffi;
+    pgp_key_search_t locator;
+    pgp_key_t *      pub;
+    pgp_key_t *      sec;
+};
```
[the remark this answers is not in the document]
Comment (ni4) [reply src/tests/key-unlock.cpp]: @dewyatt Agree on that, thanks. Pushed a commit.

**Pull request merged**